cmd/aws-actuator/main.go

```diff
@@ -145,7 +149,203 @@ func init() {
 			}
 			return nil
 		},
-	})
+	}
+}
+
+func readMachineManifest(manifestLoc string) (*clusterv1.Machine, error) {
+	machine := &clusterv1.Machine{}
+	bytes, err := ioutil.ReadFile(manifestLoc)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read %v: %v", manifestLoc, err)
+	}
+
+	if err = yaml.Unmarshal(bytes, &machine); err != nil {
+		return nil, fmt.Errorf("Unable to unmarshal %v: %v", manifestLoc, err)
+	}
+
+	return machine, nil
+}
+
+func readClusterManifest(manifestLoc string) (*clusterv1.Cluster, error) {
+	cluster := &clusterv1.Cluster{}
+	bytes, err := ioutil.ReadFile(manifestLoc)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read %v: %v", manifestLoc, err)
+	}
+
+	if err = yaml.Unmarshal(bytes, &cluster); err != nil {
+		return nil, fmt.Errorf("Unable to unmarshal %v: %v", manifestLoc, err)
+	}
+
+	return cluster, nil
+}
+
+func readSecretManifest(manifestLoc string) (*apiv1.Secret, error) {
+	secret := &apiv1.Secret{}
+	bytes, err := ioutil.ReadFile(manifestLoc)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read %v: %v", manifestLoc, err)
+	}
+	if err = yaml.Unmarshal(bytes, &secret); err != nil {
+		return nil, fmt.Errorf("Unable to unmarshal %v: %v", manifestLoc, err)
+	}
+	return secret, nil
+}
+
+const workerUserDataBlob = `#!/bin/bash
+
+cat <<HEREDOC > /root/user-data.sh
+#!/bin/bash
+
+cat <<EOF > /etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes
+baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+exclude=kube*
+EOF
+setenforce 0
+yum install -y kubelet kubeadm --disableexcludes=kubernetes
+
+cat <<EOF > /etc/default/kubelet
+KUBELET_KUBEADM_EXTRA_ARGS=--cgroup-driver=systemd
+EOF
+
+kubeadm join {{ .MasterIp }}:8443 --token 2iqzqm.85bs0x6miyx1nm7l --discovery-token-unsafe-skip-ca-verification
+
+HEREDOC
+
+bash /root/user-data.sh > /root/user-data.logs
```

Comment:
Cannot we just execute above commands instead of creating bash script `user-data.sh` just so we can later execute it? If logs are important, we can just use `>>` to append to log file.

Reply:
@frobware could tell more how useful that is :)

In case something goes wrong you might want to re-run the shell script without creating a new instance. Anyway, it's still an example and only for demonstration. Not meant to be run in production.

Reply:
Yeah, I found it useful to rerun.

But I think we also have to be careful -- there's limited value for absolute perfection here as they're really examples and will disappear over time.